src/main/scala/org/broadinstitute/dsde/workbench/sam/dataAccess/PostgresAccessPolicyDAO.scala

```diff
@@ -1350,7 +1350,59 @@ class PostgresAccessPolicyDAO(protected val writeDbRef: DbReference, protected v
     })
   }
 
-  private def setPolicyIsPublicInternal(policyPK: PolicyPK, isPublic: Boolean)(implicit session: DBSession): Int = {
+  override def recreateEffectivePolicyRolesTableEntry(resourceTypeName: ResourceTypeName, samRequestContext: SamRequestContext): IO[Unit] = {
+    val resource = ResourceTable.syntax("resource")
+    val policyResource = ResourceTable.syntax("policyResource")
+    val resourceRole = ResourceRoleTable.syntax("resourceRole")
+    val flattenedRole = FlattenedRoleMaterializedView.syntax("flattenedRole")
+    val resourceTypeOne = ResourceTypeTable.syntax("resourceTypeOne")
+    val resourceTypeTwo = ResourceTypeTable.syntax("resourceTypeTwo")
+    val policy = PolicyTable.syntax("policy")
+    val policyRole = PolicyRoleTable.syntax("policyRole")
+    val effectivePolicyRole = EffectivePolicyRoleTable.syntax("effectivePolicyRole")
+    val effectiveResourcePolicy = EffectiveResourcePolicyTable.syntax("effectiveResourcePolicy")
+
+    serializableWriteTransaction("recreateEffectivePolicyRolesTableEntry", samRequestContext)({ implicit session =>
+      samsql"""delete from ${EffectivePolicyRoleTable as effectivePolicyRole}
+               using ${EffectiveResourcePolicyTable as effectiveResourcePolicy},
+               ${ResourceRoleTable as resourceRole},
+               ${PolicyTable as policy},
+               ${ResourceTable as policyResource},
+               ${ResourceTypeTable as resourceTypeOne},
+               ${ResourceTypeTable as resourceTypeTwo}
+               where ${effectivePolicyRole.effectiveResourcePolicyId} = ${effectiveResourcePolicy.id}
+               and ${effectivePolicyRole.resourceRoleId} = ${resourceRole.id}
+               and ${effectiveResourcePolicy.sourcePolicyId} = ${policy.id}
+               and ${policy.resourceId} = ${policyResource.id}
+               and ${policyResource.resourceTypeId} = ${resourceTypeOne.id}
+               and ${resourceRole.resourceTypeId} = ${resourceTypeTwo.id}
+               and (
+                 ${resourceTypeOne.name} = ${resourceTypeName}
+                 or ${resourceTypeTwo.name} = ${resourceTypeName}
+               )
+            """.update().apply()
+
+      samsql"""insert into ${EffectivePolicyRoleTable.table}(${EffectivePolicyRoleTable.column.effectiveResourcePolicyId}, ${EffectivePolicyRoleTable.column.resourceRoleId})
```

Comment:
This is mostly from here: https://github.com/broadinstitute/sam/blob/develop/src/main/resources/org/broadinstitute/dsde/sam/liquibase/changesets/20201123_effective_policy.xml#L79